build/LocalizationTasks/SaveXlfToResx.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using xlflib;
+
+namespace Microsoft.Build.LocalizationTasks
+{
+    public class SaveXlfToResx : Task
+    {
+        [Required]
+        public string ResxPath { get; set; }
+
+        [Required]
+        public string XlfPath { get; set; }
+
+        /// <summary>
+        /// When set to "true" it makes the task replace the resource values with another value.
+        /// Useful for testing localized builds.
+        /// 
+        /// If <see cref="TestString"/> is non empty, resource values get replaced with it.
+        /// If it is empty, resource values get replaced with !resource_id!english_resource!localized_resource!
+        /// </summary>
+        public string ReplaceWithTestString { get; set; }
+
+        public string TestString { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                var xlfDocument = new XlfDocument(XlfPath);
+
+                if (!string.IsNullOrEmpty(ReplaceWithTestString) && ReplaceWithTestString.Equals("true"))
+                {
+                    ReplaceResourceValuesWithTestString(xlfDocument);
+                }
+
+                xlfDocument.SaveAsResX(ResxPath);
+
+                return true;
+            }
+            catch (Exception e)
```

Comment:
Good question. My reasoning was on the lines of: "I have no idea what exceptions might get transitively thrown from xliff parser, so then instead of letting a potential exception break out of the task, I'd rather log it as an error and have the task return false".